rfcs/20230517-forward-compatibility-testing.md

```diff
@@ -0,0 +1,99 @@
+# StableHLO Forward Compatibility Testing
+
+## Background
+
+StableHLO needs to verify both forward and backward compatibility. Currently we
+check backward compatibility by versioning and serializing
+`stablehlo_legalize_to_vhlo.mlir`, and need something similar for forward
+compatibility. This document makes the following proposal:
+
+**Proposal:** Add tests to compare `stablehlo_legalize_to_vhlo.0_X_0.mlir`
+serialized at HEAD with existing [`stablehlo_legalize_to_vhlo.0_X_0.mlir.bc`](https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/stablehlo_legalize_to_vhlo.0_9_0.mlir.bc)
+test files.
+
+## Requirements
+
+Forward compatibility testing has the following requirements, to ensure that it
+is both useful and practical:
+
+* **R1.** Tests must detect changes in StableHLO that break forward compatibility.
+* **R2.** Tests must detect LLVM revision bumps that break forward compatibility.
+* **R3.** Tests must scale with releases and execute in a reasonable amount
+of time.
+
+## Proposed Design: Statically Test Forward Compatibility
+
+In this design, forward incompatibilities are detected by serializing
+`stablehlo_legalize_to_vhlo.mlir` at head and comparing it to a known-good file
```

Comment:
Do you mean "serializing `stablehlo_legalize_to_vhlo.0_X_0.mlir` at HEAD"? (Apologies, I know I asked for the current wording in my previous comment, but I didn't fully understand the proposal at that point).

Reply:
Yes, changed to that wording.